### PR TITLE
Create github action for make and test

### PR DESCRIPTION
**Pull request by jsuchome**
https://github.com/fuseml/fuseml/issues/72

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: Go
```
Comment (flaviodsr) [inline .github/workflows/go.yml:1]: I think a better name for this would be `CI`?

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        go-version: 1.16
+        
+    - name: Install Protoc
+      uses: arduino/setup-protoc@v1
+      
+    - name: Install dependencies
+      run: |
+        go get google.golang.org/grpc/cmd/protoc-gen-go-grpc
+        go get github.com/golang/protobuf/protoc-gen-go   
+        go get golang.org/x/lint/golint
```
Comment (flaviodsr) [inline .github/workflows/go.yml:28]: `make deps` already includes part of this, maybe makes sense including what is missing and call it here?

You could also consider using https://github.com/actions/cache for caching the go packages.

**Review by flaviodsr — CHANGES_REQUESTED** [review]
Looks good, but I have some suggestions.

I see that apart from the last step both jobs are the same, in that way you could use https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix to avoid the repetition.

File: .github/workflows/go.yml (hunk 2 above)
Comment (jsuchome) [reply .github/workflows/go.yml:28]: ok

Comment (jsuchome) [reply .github/workflows/go.yml]: I added make deps call, but I do not see how the cache would actually help ...

**Comment by jsuchome** [comment]
Oh, I missed your first comment about the matrix, I'll have to look at it ...

File: .github/workflows/go.yml (hunk 2 above)
Comment (flaviodsr) [reply .github/workflows/go.yml]: Caching the go packages would make the next runs faster as they would need to download only the outdated packages, if any.

**Review by stefannica — APPROVED** [review]

**Pull request merged**